addons/web/static/src/js/libs/moment.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * This file aim to contains the update/fix we have to do on moment.js localizations.
+ * By using updateLocale and/or defineLocale here, we avoid adding change in the official
+ * moment.js files, which could lead to conflict when updating the library.
+ */
+
+odoo.define('web.moment.extensions', function () {
+'use strict';
+const locale = moment.locale();
+moment.updateLocale('ca', {
+    preparse: function (string) {
+        return string.replace(/\b(?:d’|de )(gener|febrer|març|abril|maig|juny|juliol|agost|setembre|octubre|novembre|desembre)/g, '$1');
+    }
+});
+if(locale !== 'ca'){
+    moment.locale(locale);
+}
```

Comment:
Maybe we could update Catalan localization only when it's used?
```suggestion
if(locale === 'ca'){
    moment.updateLocale('ca', {
        preparse: function (string) {
            return string.replace(/\b(?:d’|de )(gener|febrer|març|abril|maig|juny|juliol|agost|setembre|octubre|novembre|desembre)/g, '$1');
        }
    });
}
```

Reply:
Indeed, this was my first try, but (if m'y understanding is correct) this part of the code is only executed when starting the server, meaning that if someone start with the English language and then change its language to Catalan, this won't work until he restarts the server.

Reply:
Why would you need to restart the server? You probably mean refresh web page? But it's already mandatory action on switching to another language